### PR TITLE
8 creating test structure

### PR DESCRIPTION
**Pull request by tib-tib**

### CODE REVIEW

**Comment by tib-tib** [comment]
@guilhermevrs Hey dude could you review my code please ? Thanks :)

File: package.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
     "webdriver-start": "webdriver-manager start",
     "preprotractor": "npm run webdriver-update",
     "e2e": "protractor",
-    "ci": "npm run e2e && npm run test",
-    "test": "karma start",
+    "test": "npm run e2e && npm run unit",
```
Comment (guilhermevrs) [inline package.json]: I have tried to run this one but I have an error realted to the selenium server jar. Is it supposed to be like this?

File: karma.conf.js [hunk 2]
```diff
@@ -80,7 +78,7 @@ module.exports = function(config) {
 
     // level of logging
     // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
-    logLevel: config.LOG_INFO,
+    logLevel: config.LOG_WARN,
 
 
     // enable / disable watching file and executing tests whenever any file changes
```
Comment (guilhermevrs) [inline karma.conf.js]: I believe we could have autoWatch true. Or, at least an option in the scripts of npm. This way would make our lives easy :dancer:

**Comment by guilhermevrs** [comment]
Very good PR Tib! Some small stuff that we should agree (I commented them up here) and then we can merge

File: package.json (hunk 1 above)
Comment (tib-tib) [inline package.json]: Well for e2e tests you have to launch the preprotractor task before... It's weird maybe I can change that

File: karma.conf.js (hunk 2 above)
Comment (tib-tib) [inline karma.conf.js]: I agree I will add a command to run tests with autowatch and thus we will be able to do TDD :)

File: package.json (hunk 1 above)
Comment (tib-tib) [inline package.json]: @guilhermevrs and your app needs to be running if you want e2e tests to work

**Comment by tib-tib** [comment]
@guilhermevrs I have made changes according to your points, does it seem good to you now ?

**Comment by guilhermevrs** [comment]
I am not sure why I am still facing some problems regarding the e2e test. Now when I launch it, it gets stuck a little bit at:

    Selenium standalone server started at http://192.168.0.20:44199/wd/hub

and then I get this error message

    /media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/atoms/error.js:108
      var template = new Error(this.message);
                     ^
    UnknownError: unknown error: Chrome failed to start: crashed
      (Driver info: chromedriver=2.19.346067     (6abd8652f8bc7a1d825962003ac88ec6a37a82f1),platform=Linux 3.13.0-37-generic x86_64)     (WARNING: The server did not provide any stacktrace information)
    Command duration or timeout: 60.14 seconds
    Build info: version: '2.47.1', revision: '411b314', time: '2015-07-30 03:03:16'
    System info: host: 'guilhermevrs-linux', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-37-generic', java.version: '1.7.0_79'
    Driver info: org.openqa.selenium.chrome.ChromeDriver
        at new bot.Error (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/atoms/error.js:108:18)
        at Object.bot.response.checkResponse     (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/atoms/response.js:109:9)
        at /media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:160:24
        at [object Object].promise.ControlFlow.runInFrame_ (eval at <anonymous> (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/goog/base.js:1124:19), <anonymous>:1857:20)
        at [object Object].goog.defineClass.notify (eval at <anonymous> (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/goog/base.js:1124:19), <anonymous>:2448:25)
        at [object Object].promise.Promise.notify_ (eval at <anonymous> (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/goog/base.js:1124:19), <anonymous>:564:12)
        at Array.forEach (native)
        at [object Object].promise.Promise.notifyAll_ (eval at <anonymous> (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/goog/base.js:1124:19), <anonymous>:553:15)
        at [object Object].goog.async.run.processWorkQueue [as _onTimeout] (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/goog/async/run.js:130:15)
        at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)
    From: Task: WebDriver.createSession()
        at Function.webdriver.WebDriver.acquireSession_ (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:157:22)
        at Function.webdriver.WebDriver.createSession (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:131:30)
        at [object Object].Builder.build (/media/guilhermevrs/Data/Fontes/donna/node_modules/selenium-webdriver/builder.js:445:22)
        at [object Object].DriverProvider.getNewDriver (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/lib/driverProviders/driverProvider.js:38:7)
        at [object Object].Runner.createBrowser (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/lib/runner.js:182:37)
        at driverprovider_.setupEnv.then.then.then.frameworkPath (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/lib/runner.js:263:21)
        at _fulfilled (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/node_modules/q/q.js:797:54)
        at self.promiseDispatch.done (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/node_modules/q/q.js:826:30)
        at Promise.promise.promiseDispatch (/media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/node_modules/q/q.js:759:13)
        at /media/guilhermevrs/Data/Fontes/donna/node_modules/protractor/node_modules/q/q.js:573:44
[launcher] Process exited with error code 1

**Comment by tib-tib** [comment]
Hmm that's weird, have you tried removing your node_modules and reinstalling it again ? Normally e2e tests should work because it works on CircleCI, which is the hardest part ;)

**Comment by guilhermevrs** [comment]
Yes you were right. I deleted node_modules and then he changed the error message to say that my chrome was not up to date. I updated it.

The tests ran, but all of them failed. I didn't know I had to start the server before running the e2e. Should we change something here or leave like that and add a small instruction into the README.md?

**Comment by tib-tib** [comment]
I know, I am always right ;)
Well I think this is a really particular case, and actually we're not sure we'll continue doing e2e tests so my recommendation is currently not to change anything about e2e tests but sure we can update the README.md. Do you want to do it or should I do it ?

**Comment by guilhermevrs** [comment]
Do it yourself bro :)
And yes, I am not sure we are going to start the e2e tests, perhaps only in very specific cases

**Comment by tib-tib** [comment]
@guilhermevrs Done

**Comment by guilhermevrs** [comment]
Well done!

**Pull request merged**